src/ol_infrastructure/infrastructure/grafana_cloud/__main__.py

```diff
@@ -0,0 +1,73 @@
+from pulumi import Config
+from pulumi_aws import iam
+
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.vault import setup_vault_provider
+
+if Config("vault_server").get("env_namespace"):
+    setup_vault_provider()
+
+# Reference: https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/
+raw_policy = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "AllowReadingMetricsFromCloudWatch",
+            "Effect": "Allow",
+            "Action": [
+                "cloudwatch:DescribeAlarmsForMetric",
+                "cloudwatch:DescribeAlarmHistory",
+                "cloudwatch:DescribeAlarms",
+                "cloudwatch:ListMetrics",
+                "cloudwatch:GetMetricData",
+                "cloudwatch:GetInsightRuleReport",
+            ],
+            "Resource": "*",
+        },
+        {
+            "Sid": "AllowReadingLogsFromCloudWatch",
+            "Effect": "Allow",
+            "Action": [
+                "logs:DescribeLogGroups",
+                "logs:GetLogGroupFields",
+                "logs:StartQuery",
+                "logs:StopQuery",
+                "logs:GetQueryResults",
+                "logs:GetLogEvents",
+            ],
+            "Resource": "*",
+        },
+        {
+            "Sid": "AllowReadingTagsInstancesRegionsFromEC2",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeTags",
+                "ec2:DescribeInstances",
+                "ec2:DescribeRegions",
+            ],
+            "Resource": "*",
+        },
+        {
+            "Sid": "AllowReadingResourcesForTags",
+            "Effect": "Allow",
+            "Action": "tag:GetResources",
+            "Resource": "*",
+        },
+    ],
+}
+grafana_cloudwatch_policy = iam.Policy(
+    "create-grafana-readonly-cloudwatch-policy",
+    name="allow-grafana-cloud-cloudwatch-access",
+    path="/ol-operations/global-policies/",
+    policy=lint_iam_policy(
+        raw_policy, parliament_config={"RESOURCE_STAR": {}}, stringify=True
```

Comment:
Rather than ignore the `RESOURCE_STAR` linting error it would be helpful to make a pass of scoping the resources so that it's easier to visually understand what objects will be operated on.